haystack_experimental/core/pipeline/pipeline.py

```diff
@@ -99,7 +138,12 @@ def _run_component(
             return cast(Dict[Any, Any], component_output)
 
     def run(  # noqa: PLR0915, PLR0912
-        self, data: Dict[str, Any], include_outputs_from: Optional[Set[str]] = None
+        self,
+        data: Dict[str, Any],
+        include_outputs_from: Optional[Set[str]] = None,
+        breakpoints: Optional[Set[Tuple[str, Optional[int]]]] = None,
+        resume_state: Optional[Dict[str, Any]] = None,
```

Comment:
if `breakpoints` and `resume_state` are both set, let's show at least a warning that `breakpoints` will be ignored (if that's the behavior)